File: app/src/main/java/com/sevtinge/cemiuiler/module/systemframework/FreeFormCount.java
package com.sevtinge.cemiuiler.module.systemframework;

import com.sevtinge.cemiuiler.module.base.BaseHook;

import de.robv.android.xposed.XC_MethodReplacement;

public class FreeFormCount extends BaseHook {

    Class<?> mMiuiFreeFormActivityStack;
    Class<?> mMiuiFreeFormStackDisplayStrategy;
    Class<?> mMiuiFreeFormManagerService;

    @Override
    public void init() {

        mMiuiFreeFormActivityStack = findClassIfExists("com.android.server.wm.MiuiFreeFormActivityStack");
        mMiuiFreeFormStackDisplayStrategy = findClassIfExists("com.android.server.wm.MiuiFreeFormStackDisplayStrategy");
        mMiuiFreeFormManagerService = findClassIfExists("com.android.server.wm.MiuiFreeFormManagerService");

        findAndHookMethod(mMiuiFreeFormStackDisplayStrategy,"getMaxMiuiFreeFormStackCount", String.class, mMiuiFreeFormActivityStack, XC_MethodReplacement.returnConstant(100));

        findAndHookMethod(mMiuiFreeFormStackDisplayStrategy,"getMaxMiuiFreeFormStackCountForFlashBack", String.class, boolean.class, XC_MethodReplacement.returnConstant(100));

        findAndHookMethod(mMiuiFreeFormManagerService, "shouldStopStartFreeform", String.class, XC_MethodReplacement.returnConstant(false));
    }
}

File: app/src/main/java/com/sevtinge/cemiuiler/module/systemframework/FreeFormCount.kt
package com.sevtinge.cemiuiler.module.systemframework

import com.github.kyuubiran.ezxhelper.ClassUtils.loadClass
import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHook
import com.github.kyuubiran.ezxhelper.finders.MethodFinder.`-Static`.methodFinder
import com.sevtinge.cemiuiler.module.base.BaseHook

class FreeFormCount : BaseHook() {
    override fun init() {
        // GetMaxMiuiFreeFormStackCount
        loadClass("com.android.server.wm.MiuiFreeFormStackDisplayStrategy").methodFinder().first {
            name == "getMaxMiuiFreeFormStackCount"
        }.createHook {
            returnConstant(256)
        }

        // GetMaxMiuiFreeFormStackCountForFlashBack
        loadClass("com.android.server.wm.MiuiFreeFormStackDisplayStrategy").methodFinder().first {
            name == "getMaxMiuiFreeFormStackCountForFlashBack"
        }.createHook {
            returnConstant(256)
        }

        // ShouldStopStartFreeform
        loadClass("com.android.server.wm.MiuiFreeFormManagerService").methodFinder().first {
            name == "shouldStopStartFreeform"
        }.createHook {
            returnConstant(false)
        }
    }
}